models/rate_neuron_ipn_impl.h

```diff
@@ -247,21 +268,38 @@ nest::rate_neuron_ipn< TGainfunction >::update_( Time const& origin,
     // get noise
     S_.noise_ = P_.std_ * B_.random_numbers[ lag ];
     // propagate rate to new time step (exponential integration)
-    S_.rate_ = V_.P1_ * S_.rate_ + V_.P2_ * P_.mean_
+    S_.rate_ = V_.P1_ * new_rates[ lag ] + V_.P2_ * P_.mean_
       + V_.input_noise_factor_ * S_.noise_;
 
     if ( called_from_wfr_update ) // use get_value_wfr_update to keep values in
                                   // buffer
     {
       if ( P_.linear_summation_ )
       {
-        S_.rate_ += V_.P2_ * gain_( B_.delayed_rates_.get_value_wfr_update(
-                                      lag ) + B_.instant_rates_[ lag ] );
+        S_.rate_ += V_.P2_
```

Comment:
@ddahmen That is a really good catch! Of course you are right about this - the new version as is would be inconsistent to NEST 2.14 for `linear_summation = True` with a nonlinear model with both inhibitory and excitatory inputs and it even disables the use of a "joint linear summation".

@jschuecker @MSenden We will need to think about how to resolve this in more detail.

Reply:
@ddahmen Mario and I resolved this issue by moving the check for multiplicative coupling to the base class in ff74e56. Now everything should be consistent with NEST 2.14 again.

The diff of ff74e56 might look a bit confusing, since I rearranged some parts of the code to avoid code duplication (it is easier to follow the changes by looking at the new file).